src/index.js

```diff
@@ -0,0 +1,27 @@
+import { waitForPromise } from '@ember/test-waiters';
+
+export async function wrappedFetch(...args) {
+  let responsePromise = fetch(...args);
+
+  waitForPromise(responsePromise);
+
+  let response = await responsePromise;
+
+  return new Proxy(response, {
+    get(target, prop, receiver) {
+      let original = Reflect.get(target, prop, receiver);
+
+      if (['json', 'text', 'arrayBuffer', 'blob', 'formData'].includes(prop)) {
+        return (...args) => {
+          let parsePromise = original(...args);
+
+          return waitForPromise(parsePromise);
+        };
+      }
```

Comment:
This didn't work in Safari for our application. It turns out that's because some properties use "internal slots" for property access, so you have to wrangle the `this` value to be the right thing. I lifted the implementation from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#no_private_property_forwarding

```suggestion
      let original = target[prop];

      if (typeof original === 'function') {
        return function (...args) {
          let maybePromise = original.apply(this === receiver ? target : this, args);
          // only wait for promise if it is a promise
          if (maybePromise && typeof maybePromise.then === 'function') {
            return waitForPromise(maybePromise);
          }
          
          return maybePromise;
        };
      }
```

Since there are other properties like `toString` etc which can get accessed, I made it only `waitForPromise` if we have a promise-like return value as well.

Reply:
The conditional use of waitForPromise makes it so it only gets called on promises. if waitForPromise is called with a non-promise, it seems to error and hang the test suite with a waiter that never resolves.

Reply:
rfc has it as 
```js
let parsePromise = original.call(target, ...args);
```
which I've successfully used